### PR TITLE
feat: support route index and nested url without nested layout

### PR DESCRIPTION
**Pull request by luhc228**
1. Ref: https://github.com/alibaba/ice/issues/5880
2. 支持创建 `a.b.c.tsx` 扁平文件名代替 `a/b/c.tsx`嵌套目录 来生成 `/a/b/c` 嵌套 url。Ref: https://remix.run/docs/en/main/guides/routing#nested-urls-without-nesting-layouts

### CODE REVIEW

File: packages/route-manifest/src/index.ts [hunk 1]
```diff
@@ -215,17 +215,55 @@ function defineConventionalRoutes(
...
+
   const partialRouteId = removeLastLayoutStrFromId(routeId);
 
   for (let i = 0; i < partialRouteId.length; i++) {
     const char = partialRouteId.charAt(i);
-
+    const lastChar = i > 0 ? partialRouteId.charAt(i - 1) : undefined;
     const nextChar = i < partialRouteId.length - 1 ? partialRouteId.charAt(i + 1) : undefined;
 
+    function isNewEscapeSequence() {
```
Comment (ClarkXia) [inline packages/route-manifest/src/index.ts:235]: 函数声明尽量不要在循环体内

**Comment by ClarkXia** [comment]
test case 需要覆盖新规则

File: packages/route-manifest/src/index.ts [hunk 2]
```diff
@@ -215,17 +215,55 @@ function defineConventionalRoutes(
...
+
+    if (skipSegment) {
+      if (char === '/' || char === '.' || char === path.win32.sep) {
+        skipSegment = false;
+      }
+      continue;
+    }
+
+    if (isNewEscapeSequence()) {
+      inEscapeSequence++;
```
Comment (chenjun1011) [inline packages/route-manifest/src/index.ts:253]: 建议加个例子，说明下处理的场景

File: packages/route-manifest/src/index.ts (hunk 1 above)
Comment (luhc228) [reply packages/route-manifest/src/index.ts]: fixed

**Comment by luhc228** [comment]
> test case 需要覆盖新规则

done

File: packages/route-manifest/src/index.ts (hunk 2 above)
Comment (luhc228) [reply packages/route-manifest/src/index.ts:246]: done

**Review by ClarkXia — APPROVED** [review]

**Review by chenjun1011 — APPROVED** [review]

**Pull request merged**